core/src/utils/mod.rs

```diff
@@ -135,3 +136,24 @@ fn debug_assert_is_checked() {
     // https://github.com/0xPolygonMiden/miden-vm/issues/433
     debug_assert!(false);
 }
+
+// FORMATTING
+// ================================================================================================
+
+/// Utility to write a sequence of bytes as hex to a Formatter.
+pub fn to_hex<T, I>(bytes: T) -> Result<String, fmt::Error>
+where
+    T: IntoIterator<Item = I>,
+    I: fmt::LowerHex,
+{
+    let bytes = bytes.into_iter();
+    let hint = bytes.size_hint();
+    let size = hint.1.unwrap_or(hint.0);
+    let mut s = String::with_capacity(size * 2);
+
+    for byte in bytes {
+        write!(s, "{byte:02x}")?;
+    }
+
+    Ok(s)
+}
```

Comment:
nit: I think the comment needs to be updated.

Also, I wonder this is a bit more complicated than what we need. Why not take a simple `&[u8]` as a parameter? Then the whole thing would look like:
```Rust
let mut s = String::with_capacity(bytes.len() * 2);
for byte in bytes {
    write!(s, "{byte:02x}")?;
}
Ok(s)
```
But I'm fine either way.

Reply:
> Also, I wonder this is a bit more complicated than what we need. Why not take a simple &[u8] as a parameter? Then the whole thing would look like:

It was just to make the caller side easier to write. Reverted it.

Reply:
> nit: I think the comment needs to be updated.

nice caught, fixed!